R/case-variables.R

```diff
@@ -0,0 +1,149 @@
+#' Case statement for case variables
+#'
+#' List elements: 
+#' id an integer to use for this category when a case variable is made 
+#' (default: none, one will automatically be assigned when the case variable 
+#' is made)
+#' name a charcater identifier for this case
+#' expression `CrunchLogicalExpr` which sets the conditions for the case
+#' numeric_value a numeric which is the value this case should take on 
+#' (useful when made into a case variable)
+#' missing a logical indicating if this case should be treated as missing
+#' 
+#' @keywords internal
+ensureValidCases <- function(case) {
+    if (!is.list(case)) {
+        halt("A case must be a list")
+    }
+    wrong_case_names <- setdiff(names(case), c("id", "name", "expression",
+                                               "numeric_value", "missing"))
+    if (length(wrong_case_names) > 0) {
+        halt("each case must have at most an id, name, expression, numeric_value, ", 
```

Comment:
You could `serialPaste(wrong_case_names)` in the error message so the user can see what they provided wrong. I've found that helpful in the past.